hudi-client/src/main/java/org/apache/hudi/table/action/rollback/BaseRollbackActionExecutor.java

```diff
@@ -159,24 +161,32 @@ private void rollBackIndex() {
     LOG.info("Index rolled back for commits " + instantToRollback);
   }
 
-  public List<HoodieRollbackStat> doRollbackAndGetStats() {
-    final String instantTimeToRollback = instantToRollback.getTimestamp();
-    final boolean isPendingCompaction = Objects.equals(HoodieTimeline.COMPACTION_ACTION, instantToRollback.getAction())
-        && !instantToRollback.isCompleted();
-    validateSavepointRollbacks();
-    if (!isPendingCompaction) {
-      validateRollbackCommitSequence();
-    }
-
-    try {
-      List<HoodieRollbackStat> stats = executeRollback();
-      LOG.info("Rolled back inflight instant " + instantTimeToRollback);
+  public List<HoodieRollbackStat> mayBeRollbackAndGetStats(boolean doDelete) {
+    if(doDelete) {
+      final String instantTimeToRollback = instantToRollback.getTimestamp();
+      final boolean isPendingCompaction = Objects.equals(HoodieTimeline.COMPACTION_ACTION, instantToRollback.getAction())
```

Comment:
@vinothchandar : forgot to remind you yesterday when we discussed to move the collectStats method to a separate class and call directly for upgrade. These validations steps (validateSavepointRollbacks, validateRollbackCommitSequence) might be required as well right ? So, few bits and pieces in this class is required in upgrade step as well.